static/js/components/widgets/ResourcePickerDialog.tsx

```diff
@@ -5,42 +5,82 @@ import Dialog from "../Dialog"
 import {
   RESOURCE_TYPE_DOCUMENT,
   RESOURCE_TYPE_IMAGE,
-  RESOURCE_TYPE_VIDEO
+  RESOURCE_TYPE_VIDEO,
+  CONTENT_TYPE_RESOURCE,
+  CONTENT_TYPE_PAGE
 } from "../../constants"
 import ResourcePickerListing from "./ResourcePickerListing"
 import { useDebouncedState } from "../../hooks/state"
 import {
   CKEResourceNodeType,
-  ResourceDialogState,
-  RESOURCE_EMBED
+  ResourceDialogMode,
+  RESOURCE_EMBED,
+  RESOURCE_LINK
 } from "../../lib/ckeditor/plugins/constants"
 import { WebsiteContent } from "../../types/websites"
 
 interface Props {
-  state: ResourceDialogState
+  mode: ResourceDialogMode
+  isOpen: boolean
```

Comment:
Previously the state variable had values roughly `["link", "embed", "closed"]. Using a single prop for both "content/behavior" state and "open/closed" state meant we could not close the modal without changing its content. See gif below (which is from rc). This was a pretty minor visual bug but became more noticeable with the addition of "Pages" to the Link dialog variant (since now the Link/Embed variants are more different).

![modal_contents_change](https://user-images.githubusercontent.com/9010790/150430654-e35e4a7e-3456-4f1c-b153-39782a0fb5a8.gif)

Reply:
one other option for resolving the funny UI flash could be to hide the button in the case that the user has already clicked 'cancel'...which, come to think of it, we might want to do anyway.

Thinking about this made me wonder if it was possible to still click the 'link resource' button _after_ clicking 'cancel' and while you have to have pretty good timing I was able to do so. So we might want to hide the 'submit' button after cancel is hit.

what do you think?

Reply:
BTW: Your question about whether the button should disappear when clicking close made me wonder how other sites handle this, so I spent a few minutes looking. I checked a few sites, of which only Linkedin and Trello had modals thats. Both Linkedin and Trello make their modals disappear immediately (no transition). Now I will be on the lookout for sites with transitiony modals, and how they handle this. 🤷 

But... getting rid of the transition definitely stuff like this easier to avoid, so that could be another possibility.

Reply:
yeah, I think I'd prefer no transition but we might be a bit stuck with it

it might be a good idea to clear the selection when the tab changes, but I don't think we need to do so in this PR

and yeah, that makes sense, I think we can just leave the state as is